test/type.js

```diff
@@ -54,4 +54,206 @@ describe('type(obj)', function () {
     assert('object' === type(new String('hello')));
   });
+
+  describe('New ECMA6 Types Stubbed', function () {
+    var originalObjectToString = Object.prototype.toString;
+
+    function stubObjectToStringOnce(staticValue) {
+      Object.prototype.toString = function () {
+        Object.prototype.toString = originalObjectToString;
+        return staticValue;
+      };
+    }
+
+    it('map', function () {
+      stubObjectToStringOnce('[object Map]');
+      assert('map' === type({}));
+    });
+
+    it('weakmap', function () {
+      stubObjectToStringOnce('[object WeakMap]');
+      assert('weakmap' === type({}));
+    });
+
+    it('set', function () {
+      stubObjectToStringOnce('[object Set]');
+      assert('set' === type({}));
+    });
+
+    it('weakset', function () {
+      stubObjectToStringOnce('[object WeakSet]');
+      assert('weakset' === type({}));
+    });
+
+    it('symbol', function () {
+      stubObjectToStringOnce('[object Symbol]');
+      assert('symbol' === type({}));
+    });
+
+    it('promise', function () {
+      stubObjectToStringOnce('[object Promise]');
+      assert('promise' === type({}));
+    });
+
+    it('int8array', function () {
+      stubObjectToStringOnce('[object Int8Array]');
+      assert('int8array' === type({}));
+    });
+
+    it('uint8array', function () {
+      stubObjectToStringOnce('[object Uint8Array]');
+      assert('uint8array' === type({}));
+    });
+
+    it('uint8clampedarray', function () {
+      stubObjectToStringOnce('[object Uint8ClampedArray]');
+      assert('uint8clampedarray' === type({}));
+    });
+
+    it('int16array', function () {
+      stubObjectToStringOnce('[object Int16Array]');
+      assert('int16array' === type({}));
+    });
+
+    it('uint16array', function () {
+      stubObjectToStringOnce('[object Uint16Array]');
+      assert('uint16array' === type({}));
+    });
+
+    it('int32array', function () {
+      stubObjectToStringOnce('[object Int32Array]');
+      assert('int32array' === type({}));
+    });
+
+    it('uint32array', function () {
+      stubObjectToStringOnce('[object Uint32Array]');
+      assert('uint32array' === type({}));
+    });
+
+    it('float32array', function () {
+      stubObjectToStringOnce('[object Float32Array]');
+      assert('float32array' === type({}));
+    });
+
+    it('float64array', function () {
+      stubObjectToStringOnce('[object Float64Array]');
+      assert('float64array' === type({}));
+    });
+
+    it('dataview', function () {
+      stubObjectToStringOnce('[object DataView]');
+      assert('dataview' === type({}));
+    });
+
+    it('arraybuffer', function () {
+      stubObjectToStringOnce('[object ArrayBuffer]');
+      assert('arraybuffer' === type({}));
+    });
+  });
+
+  describe('New ECMA6 Types Conditional', function () {
+    if (typeof Map === 'function') {
+      it('map', function () {
+        assert('map' === type(new Map()));
+      });
+    }
```

Comment:
Can you add an `else` statement here so if the tests are not invoked they register in mocha as skipped? This would provide more visual feedback in each environment as to what was available and tested.

```js
} else {
  it('map');
}
```